### ISSUE TITLE
Full Bug Report - Latest Test Cycle - All 10 Tests

### ISSUE THREAD

**Issue by yaels818** [orellazri/ourstreets#3]
🟥 = Critical / High | 🟧 = Medium | 🟨 = Low

We should go over these and decide which bugs we can realistically fix before Delivery Day:

### Test 01
* 🟧 X button in the options menu works incosistently - you have to be very, VERY precise when tapping to get a reaction... Not user-friendly. I don't know if it's just my device or on all devices. When I tried to press home button to go back it got back out to Expo menu. 
<img width = "250" src = https://user-images.githubusercontent.com/99403939/169645897-9d7631a7-a162-4d44-80d3-1782e61edd2e.jpg >

### Test 02, 03, 07, 08, 09, 10
* 🟥 No return to main menu option after report has been sent!
Tapping phone's home button causes "sending report..." screen to re-appear. <br>

<img width = "250" src = https://user-images.githubusercontent.com/99403939/169645990-2887a22a-29aa-4bc9-83f4-17c4f3adc651.jpg > <img width = "250" src = https://user-images.githubusercontent.com/99403939/169645991-355618d1-6200-4edf-aef4-89000f4032df.jpg >

* 🟥 Wrong location in report summary - I entered "הצפצפה, חצור הגלילית" but instead got in the summary  "הבוסתן/הפלמ"ח Hatsor Haglilit, Israel".                                                                                                                                                                            <br> I tried again with "אורי אליאב 37, אשקלון" but got "Dafna St 24, Ashkelon"...                                                                                  <br> Third try with "השקד מגדל העמק" I got "HaShaked St. 11 , Migdal HaEmek" which is better :) <br>

<img width = "250" src = https://user-images.githubusercontent.com/99403939/169646065-2c3a4833-1fd6-48a5-aba6-d510d62e6d68.jpg ><img width = "250" src = https://user-images.githubusercontent.com/99403939/169646067-e946088a-77ab-4ce8-906a-5027a11913f7.jpg >

### Test 04 
* 🟧 Password field is a bit too low on the screen, the keyboard hides it when it's open. 
* 🟨 No password confirmation field - maybe we should lower the length of the password to prevent human errors?
* 🟨 No Phone number info bubble - we talked about maybe adding something like that for transparency (See issue  #1 ).

<img width = "250" src = https://user-images.githubusercontent.com/99403939/169646132-6ae42537-0c17-4d09-b140-329693828150.jpg >

### Test 05, 06
* ✅  Fixed from previous run! - Drop down menu now opens with "Welcome <Username>".

### Test 09
* 🟥 App crashed when I tapped "Send report" without location. 
(This can only be done when you first open the app, when I tried to run it as remove/edit location it simply took the location of the previous report). 

<img width = "250" src = https://user-images.githubusercontent.com/99403939/169646166-f651d58b-428a-4191-aeee-be2616c1fb24.jpg><img width="442" alt="Location Crash" src="https://user-images.githubusercontent.com/99403939/169646299-0a9111ee-4f0b-4e34-b912-7a6c3e7dfa85.png">

**Comment by orellazri** [comment]
Thanks Yael! Amazing work.
Can you please elaborate on Test 09? How were you able to send the report with no location?

**Comment by orellazri** [comment]
Regarding 01: Looking into the touch issue, but back button is fixed (see https://github.com/orellazri/ourstreets/commit/d1f8009351fd8eccb68d5a3282a5ce961b240d5e)
Regarding 02: Fixed. There is now return to menu and a share button with temporary text (until we get can a link to the published app)
Regarding 03: Removed location text from the report summary screen.
Regarding 04: In progress
Regarding 09: I wasn't able to reproduce.

**Comment by yaels818** [comment]
* 01:  Back button works great! 
If you're not feeling any touch issues with the X then it's fine, going back with the home button is good enough 👍 

* 02:  I'm not seeing any return to menu button or share.
Still seeing the exact same screen I pasted last time with "הדיווח התקבל" without any buttons.
I made sure to pull before I started, I should be in sync with the latest commit ([047116 - Remove Unnecessary log](https://github.com/orellazri/ourstreets/commit/047116dfe42ef22a85fc67bc28a541951083e41d))
<img width = 250 src = https://user-images.githubusercontent.com/99403939/170764384-fac0dd85-a249-46a5-93ad-a67275e199a1.jpg >

* 03:  Yes, but we should make sure there isn't any deeper issue, meaning that the location being saved for the report in the DB really matches the location the user typed, and not the location that was previously shown in the summary - if there isn't a match, that's a serious issue. 
I tried to check in the DB myself but Google security went all "You shall not enter!" on me, so... 

* 04:  I didn't run again in the meantime, waiting for updates 👍 

* 09:  That's weird, I tried again and got a different result from the last cycle - the app is now stuck on "שולח דיווח" instead of crashing. I am not sure if that's an improvement because users probably wouldn't know what's going on. 
Theory: maybe it didn't reproduce on your device because you gave it full location permissions and I didn't? 
My location is off but I figured it shouldn't matter because I'm typing different locations anyway. 

New things:
* A) License agreement button + screen + return button - all worked great 👍 
* B) New hazard type - the first time I tried reporting with it the type title was missing from the summary screen. 
It was just ONCE, **no need to debug for now**- I tried it again a few times, including shutting down the app and rebooting, and it didn't reproduce so I assume it's fine, but I'll keep an eye on it. 

<img width = 250 src = https://user-images.githubusercontent.com/99403939/170764441-c9042448-d276-420e-adf3-b0a9a67ba898.jpg ><img width = 250 src = https://user-images.githubusercontent.com/99403939/170764461-42c2b317-8c70-4931-b33a-6dac829ae8ec.jpg >
 
**Thank you for the hard work and sorry for this being so long!**
